### PR TITLE
Fixes #30506: Ensure selinux packages are installed before puppet run

### PR DESCRIPTION
**Pull request by ehelms**

### CODE REVIEW

File: hooks/pre/31_install_scl_packages.rb [hunk 1]
```diff
@@ -1,8 +1,16 @@
...
+  end
+
+  packages << 'foreman-selinux' if foreman_server?
+  packages << 'katello-selinux' if katello_enabled?
+  packages << 'candlepin-selinux' if katello_enabled?
   packages << 'pulpcore-selinux' if pulpcore_enabled?
-  ensure_packages(packages, 'installed')
 end
+
+ensure_packages(packages, 'installed') unless packages.empty?
```
Comment (ekohl) [inline hooks/pre/31_install_scl_packages.rb]: `ensure_packages` already has the shortcircuit, no need to repeat this:
https://github.com/theforeman/foreman-installer/blob/79f509685e75141038f03e0505018ba7e0a920de/hooks/boot/01-kafo-hook-extensions.rb#L18-L19

```suggestion
ensure_packages(packages, 'installed')
```

**Comment by ehelms** [comment]
To support the Puppet work around and allow breaking dependencies in packaging: https://github.com/theforeman/foreman-packaging/pull/5591

File: hooks/boot/01-kafo-hook-extensions.rb [hunk 2]
```diff
@@ -45,6 +45,10 @@ def foreman_server?
     module_enabled?('foreman')
   end
 
+  def katello_enabled?
+    module_enabled?('katello')
+  end
```
Comment (ekohl) [inline hooks/boot/01-kafo-hook-extensions.rb:50]: This makes me wonder if we should replace all uses of `module_enabled?('katello')` but we didn't replace other uses of `module_enabled?`

**Review by ekohl — COMMENTED** [review]
Overall :+1:, just a small nit at the end.

File: hooks/boot/01-kafo-hook-extensions.rb (hunk 2 above)
Comment (ehelms) [reply hooks/boot/01-kafo-hook-extensions.rb:50]: I will follow up with a PR not to muddy this one with more changes.

**Review by ekohl — APPROVED** [review]

**Pull request merged**